test/dotnet-test.Tests/GivenThatWeWantToRunTestsInTheConsole.cs

```diff
@@ -21,14 +21,17 @@ public GivenThatWeWantToRunTestsInTheConsole()
         {
             var testAssetManager = new TestAssetsManager(Path.Combine(RepoRoot, "TestAssets"));
             var testInstance =
-                testAssetManager.CreateTestInstance("ProjectWithTests", identifier: "ConsoleTests").WithLockFiles();
+                testAssetManager.CreateTestInstance("ProjectWithTests", identifier: "ConsoleTests");
 
             _projectFilePath = Path.Combine(testInstance.TestRoot, "project.json");
             var contexts = ProjectContext.CreateContextForEachFramework(
                 _projectFilePath,
                 null,
                 PlatformServices.Default.Runtime.GetAllCandidateRuntimeIdentifiers());
 
+            // Restore the project again in the destination to resolve projects
```

Comment:
Can these give a little bit deeper of a reason of "why"?  That way when whatever the underlying issue is fixed, this restore can be removed.